Comments: Add pings UI

Copons opened the pull request:
WIP

Discussion here: #15205

Copons commented:
@dmsnell no, but I'd really like the chance to do an UI for trocks!

megsfulton commented:
Ran into an issue where the site pinging back doesn't have a site icon and everything shifts to the left: 

<img width="897" alt="screen shot 2017-09-22 at 11 50 57 am" src="https://user-images.githubusercontent.com/161697/30759790-64568a0a-9f8c-11e7-9815-d235a9931a34.png">

Also can we make it so that the pingback post title and URL maintain the same column width as the comment author/url?

vindl commented on client/blocks/comment-detail/index.jsx:
```diff
@@ -239,9 +240,12 @@ export class CommentDetail extends Component {
 			isExpanded,
 		} = this.state;
 
+		const commentIsPing = 'pingback' === commentType || 'trackback' === commentType;
```
Nitpick: I don't find this very readable and it will be even less so when we add `'reactions'` here. :smile: We could replace it with `_.includes`.

megsfulton commented:
>On a different note, the pingback icon looks too big to me and it covers a large part of the site icon.

Agree but making it any smaller and you start to lose the shape of the icon. 

Looking at this again with fresh eyes + with the new info about pingbacks not having an author, I'm wondering if we should maybe go with a simpler route of having the pingback and reblog icons replace the site/profile icons? 

![image](https://user-images.githubusercontent.com/161697/30762379-adedfc98-9f96-11e7-86c1-4d90ab5132a3.png)

One disadvantage with that approach is that you lose the profile icon for the reblog author but that seems like a possibly okay tradeoff.

Copons commented:
> Also can we make it so that the pingback post title and URL maintain the same column width as the comment author/url?

@megsfulton not sure what you mean. Like, limiting it to roughly half the width of the comment card? Wouldn't it look odd, with the text cut off and so much empty space available on the right?

gwwar commented on client/blocks/comment-detail/comment-detail-author.jsx:
```diff
@@ -30,6 +30,7 @@ export class CommentDetailAuthor extends Component {
 		authorUsername: PropTypes.string,
 		blockUser: PropTypes.func,
 		commentDate: PropTypes.string,
+		commentIsPing: PropTypes.bool,
```
Could we pass through commentType instead, it'd be an enum of 'comment', 'ping', and could be expanded later if say reactions land 😆 

Since we also have `showAuthorInfo` I would recommend changing the value from the caller, instead of adding another case in the component. If I pass though`showAuthorInfo=true` and `commentIsPing=true` without looking at the internal details, I'd expect to see the author info.

gwwar replied:
We might also consider not passing through `commentIsPing` or `commentType` and maybe pass dumb boolean props instead like `showLinkIcon`

gwwar commented:
I left a few ignorable notes 😄 

@megsfulton were you suggesting to only show the link/reblog icon instead of user/site avatar? I'd be fine with a simple approach here.

megsfulton commented:
>Like, limiting it to roughly half the width of the comment card? Wouldn't it look odd, with the text cut off and so much empty space available on the right?

@Copons - sorry for that not being very clear! Easier to say it with pictures :) 

<img width="787" alt="screen shot 2017-09-22 at 5 06 16 pm" src="https://user-images.githubusercontent.com/161697/30768133-6c5fd2ca-9fb8-11e7-8282-77f09bde96ca.png">

You're right, it does leave a large gap on the right but I think that having the strong vertical columns makes it easier to scan through the list. 

If we felt that we needed something there to balance it out we could add in the post excerpt. 
<img width="786" alt="screen shot 2017-09-22 at 5 06 23 pm" src="https://user-images.githubusercontent.com/161697/30768168-d996ee3c-9fb8-11e7-8b1c-3b40e25a679c.png">

---

>were you suggesting to only show the link/reblog icon instead of user/site avatar? 

@gwwar - Yep, it seems like the pingback / reblog information is the more valuable information to show rather than the site's icon and still distinguishes them from comments.

Copons commented:
Updated visuals (cc @megsfulton)

<img width="743" alt="screen shot 2017-09-25 at 13 25 48" src="https://user-images.githubusercontent.com/2070010/30808389-d347641e-a1f5-11e7-96cc-60a7c5f4c4b7.png">

I've replaced the site icon for pings with the link gridicon, and re-added the ping content in collapsed view because cutting off the first column _really_ seemed too much off to me.

---

Updated props (cc @gwwar @vindl)

I've replaced `commentIsPing` with `commentType`.
I've opted to check on `'comment'` (instead of `'pingback' || 'trackback'`), at least for now.

vindl commented on client/blocks/comment-detail/comment-detail-author.jsx:
```diff
@@ -114,8 +117,8 @@ export class CommentDetailAuthor extends Component {
 	authorMoreInfo() {
-		if ( ! this.showMoreInfo() ) {
-			return;
+		if ( ! this.showMoreInfo() || 'comment' !== this.props.commentType ) {
```
The second part of the check looks redundant (the expression doesn't depend on it). It will only be evaluated when `! this.showMoreInfo()` is `false`, and for that to happen `'comment' === this.props.commentType` must be `true`.

Copons replied:
You're right! I missed this other check

vindl commented on client/blocks/comment-detail/comment-detail-author.jsx:
```diff
@@ -223,10 +227,19 @@ export class CommentDetailAuthor extends Component {
-					} } />
+					<div className="comment-detail__author-avatar">
+						<div className="comment-detail__author-avatar">
+							{ 'comment' === commentType &&
```
Nitpick (optional): maybe we could move this to `getCommentImage` (or some more appropriate name) helper function for readability. We are also doing something similar bellow in `comment-detail-header.jsx` where it could also be reused.

megsfulton commented:
Looks good, thanks for making that change!
